config-service-impl/src/main/java/org/hypertrace/config/service/store/ConfigStore.java

```diff
@@ -0,0 +1,42 @@
+package org.hypertrace.config.service.store;
+
+import com.google.protobuf.Value;
+import com.typesafe.config.Config;
+import org.hypertrace.config.service.ConfigResource;
+import org.hypertrace.config.service.v1.GetConfigResponse;
+
+import java.io.IOException;
+import java.util.Optional;
+import org.hypertrace.config.service.v1.UpsertConfigResponse;
+
+public interface ConfigStore {
+
+  /**
+   * Initialize the config store
+   *
+   * @param config
+   */
+  void init(Config config);
+
+  /**
+   * Write the config value associated with the specified config resource to the store.
+   *
+   * @param configResource
+   * @param userId
+   * @param config
+   * @return
+   */
+  UpsertConfigResponse writeConfig(ConfigResource configResource, String userId, Value config)
+      throws IOException;
+
+  /**
+   * Get the config for the specified resource with the specified version(optional). If
+   * configVersion is empty, get the config for the latest version.
+   *
+   * @param configResource
+   * @param configVersion
+   * @return
+   */
+  GetConfigResponse getConfig(ConfigResource configResource, Optional<Long> configVersion)
```

Comment:
generally we don't want to make optional part of an API - instead, overload getConfig for a version with long and version without. The reason here is that using an Optional in an api like this provides multiple different ways of omitting a value - a null or an empty optional, leaving ambiguity in the API and adding more work internally to account for more input states.

Reply:
I also think it's a bit strange to mix concerns here by returning a GetConfigResponse - the data store is now coupled to the external API. Would it make sense to return a value proto (or some new object if there's more data needed)

Reply:
Totally makes sense. Updated it.